pkg/check/datacenter_consistency.go

```
@@ -0,0 +1,86 @@
+	cfg := ctx.VMConfig.Config
+
+	for _, fd := range infra.Spec.PlatformSpec.VSphere.FailureDomains {
+		vcConfig := cfg.VirtualCenter[fd.Server]
+		if vcConfig == nil {
+			klog.V(4).Infof("vCenter %q not found in cloud provider config for failure domain %q, skipping (handled by CheckInfraConfig)", fd.Server, fd.Name)
+			continue
+		}
+
+		configuredDCs := parseDatacenters(vcConfig.Datacenters)
+		if !containsString(configuredDCs, fd.Topology.Datacenter) {
+			err := fmt.Errorf("Datacenter-Consistency: failure domain %q (infrastructure.config.openshift.io/cluster) "+
```

golangci-lint (2.11.4)

[error] ST1005: error strings should not end with punctuation or newlines

(staticcheck)